airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -42,6 +43,31 @@ class SlackStream(HttpStream, ABC):
     url_base = "https://slack.com/api/"
     primary_key = "id"
     page_size = 100
+    # We need different retry behaviour for different response types, implement
+    # own retry counter.
+    retry_attempts = 0
+    max_retry_attempts = 5
+
+    @property
+    def max_retries(self) -> Union[int, None]:
+        self.retry_attempts = 0
+        # Return None for unlimited retry attempts. This stream implements its
+        # own retry counter based on response type.
+        return None
+
+    def should_retry(self, response: requests.Response) -> bool:
```

Comment:
I'm a little worried about introducing infinite retry here. can we just raise number of retries to 20 or something?

Reply:
Yeah, you right, added time limit on 429 retry attempts